app/backend/src/couchers/servicers/events.py

```diff
@@ -1043,13 +1048,19 @@ def ListMyEvents(self, request, context, session):
             occurrences = occurrences.where(EventOccurrence.end_time < page_token + timedelta(seconds=1)).order_by(
                 EventOccurrence.start_time.desc()
             )
-
-        occurrences = occurrences.limit(page_size + 1)
+        # Count the total number of items for pagination
+        total_items = session.execute(select(func.count()).select_from(occurrences.subquery())).scalar()
+        # Apply pagination by page number
+        occurrences = (
+            occurrences.offset(offset).limit(page_size) if request.page_number else occurrences.limit(page_size + 1)
+        )
         occurrences = session.execute(occurrences).scalars().all()
 
         return events_pb2.ListMyEventsRes(
             events=[event_to_pb(session, occurrence, context) for occurrence in occurrences[:page_size]],
             next_page_token=str(millis_from_dt(occurrences[-1].end_time)) if len(occurrences) > page_size else None,
+            page_number=page_number,
+            total_items=total_items,
```

Comment:
I believe (I can be wrong here, just an possible idea)

changing the return of this endpoint, could lead to a different proto object, leading to a different type in the frontend, and maybe that could be the issue of the build failing?

Reply:
The issue ended up being that the `MyEvents` tests was throwing an error as it was missing the `pageNumber` and `totalItems` in the api response. I handled this by making these keys optional in the response in the protos. Then we can just use them if we want to, but don't have to.